R/exploratoryfactoranalysis.R

```diff
@@ -196,13 +230,57 @@ ExploratoryFactorAnalysis <- function(jaspResults, dataset, options, ...) {
   modelContainer[["bartab"]] <- bartab
 
   if (!ready) return()
-  bar <- psych::cortest.bartlett(dataset)
+
+  # Modification here:
+  # if "basedOn = mixed", bartlett's test is calculated
+  # based on the polychoric/tetrachoric matrix.
+  if (options[["basedOn"]] == "mixed") {
+    polytetracor <- psych::mixedCor(dataset)
+    bar <- psych::cortest.bartlett(polytetracor$rho, n = nrow(dataset))
+  }
+  else {
+    bar <- psych::cortest.bartlett(dataset)
+  }
 
   bartab[["chisq"]] <- bar[["chisq"]]
   bartab[["df"]]    <- bar[["df"]]
   bartab[["pval"]]  <- bar[["p.value"]]
 }
 
+# Modification here:
+# Added Mardia's tests of multivariate normality for further probing of the
+# multivariate normality assumption.
+.efaMardia <- function(modelContainer, dataset, options, ready) {
+  if (!options[["martest"]] || !is.null(modelContainer[["martab"]])) return()
+
+  martab <- createJaspTable(gettext("Mardia's tests of Multivariate Normality"))
```

Comment:
```suggestion
  martab <- createJaspTable(gettext("Mardia's Test of Multivariate Normality"))
```

Reply:
Noted! Will apply in future changes/PRs :)